nodejs/aws-infra/experimental/clusterService.ts

```diff
@@ -0,0 +1,337 @@
+//     }
+// }
+
+// export class HostPathVolume implements cloud.HostPathVolume {
+//     public readonly kind: cloud.VolumeKind;
+//     public readonly path: string;
+
+//     constructor(path: string) {
+//         this.kind = "HostPathVolume";
+//         this.path = path;
+//     }
+
+//     getVolumeName() {
+//         return utils.sha1hash(`${this.kind}:${this.path}`);
+//     }
+
+//     getHostPath() {
+//         return this.path;
+//     }
+// }
+
+// The shape we want for ClusterFileSystemArgs.  We don't export this as 'Overwrite' types are not pleasant to
+// work with. However, they internally allow us to succinctly express the shape we're trying to
+// provide. Code later on will ensure these types are compatible.
+type OverwriteShape = utils.Overwrite<utils.Mutable<aws.ecs.ServiceArgs>, {
+    cluster: mod.Cluster;
+    taskDefinition: mod.TaskDefinition;
+    desiredCount?: pulumi.Input<number>;
+    launchType?: pulumi.Input<"EC2" | "FARGATE">;
+    os?: pulumi.Input<"linux" | "windows">;
+    waitForSteadyState?: pulumi.Input<boolean>;
+    autoScalingGroup?: mod.ClusterAutoScalingGroup;
+}>;
+
+export interface ClusterServiceArgs {
+    // Properties from aws.ecs.ServiceArgs
+
+    /**
+     * The upper limit (as a percentage of the service's desiredCount) of the number of running
+     * tasks that can be running in a service during a deployment. Not valid when using the `DAEMON`
+     * scheduling strategy.
+     */
+    deploymentMaximumPercent?: pulumi.Input<number>;
+    /**
+     * The lower limit (as a percentage of the service's desiredCount) of the number of running
+     * tasks that must remain running and healthy in a service during a deployment.
+     */
+    deploymentMinimumHealthyPercent?: pulumi.Input<number>;
+    /**
+     * Seconds to ignore failing load balancer health checks on newly instantiated tasks to prevent
+     * premature shutdown, up to 7200. Only valid for services configured to use load balancers.
+     */
+    healthCheckGracePeriodSeconds?: pulumi.Input<number>;
+    /**
+     * ARN of the IAM role that allows Amazon ECS to make calls to your load balancer on your
+     * behalf. This parameter is required if you are using a load balancer with your service, but
+     * only if your task definition does not use the `awsvpc` network mode. If using `awsvpc`
+     * network mode, do not specify this role. If your account has already created the Amazon ECS
+     * service-linked role, that role is used by default for your service unless you specify a role
+     * here.
+     */
+    iamRole?: pulumi.Input<string>;
+    /**
+     * A load balancer block. Load balancers documented below.
+     */
+    loadBalancers?: pulumi.Input<pulumi.Input<{
+        containerName: pulumi.Input<string>;
+        containerPort: pulumi.Input<number>;
+        elbName?: pulumi.Input<string>;
+        targetGroupArn?: pulumi.Input<string>;
+    }>[]>;
```

Comment:
User can still provide this.  However, we'll combine whatever they provide with the data given by the loadBalancerProvider